README.md

```diff
@@ -1,94 +1,142 @@
-
-## Provided capabilities
-
-This skeleton mixin project brings some free capabilities:
-
-### File System Access and Context
-
-Porter provides a [Context](https://github.com/deislabs/porter/tree/main/pkg/context) package that has helpful mechanisms for accessing the File System using [spf13/afero](https://github.com/spf13/afero). This makes it easy to provide mock File System implementations during testing. The Context package also provides a mechanism to encapsualte stdin, stdout and stderr so that they can easily be passed from `cmd/docker` code to implementing `pkg/docker` code.  
-
-### Template and Static Asset Handling
-
-The project already includes [Packr V2](https://github.com/gobuffalo/packr/tree/master/v2) for dealing with static files, such as templates or other content that is best modeled outside of a Go file. You can see an example of this in `pkg/docker/schema.go`.
-
-### Basic Schema
-
-The project provides an implementation of the `docker schema` command that is mostly functional. To fully implement this for your mixin, you simply need to provide a valid JSON schema. For reference, consult `pkg/docker/schema/schema.json`.
-
-### Basic Tests
-
-The project provides some very basic test skeletons that you can use as a starting point for building tests for your mixin.
-
-### Makefile
-
-The project also includes a Makefile that will can be used to both build and install the mixin. The Makefile also includes a TODO `publish` target that shows how you might publish the mixin and generate an mixin feed for easily sharing your mixin.
\ No newline at end of file
+# A Docker Mixin for Porter
+
+This is a Docker mixin for Porter. The mixin provides the Docker CLI. 
+
+## Mixin Declaration
+
+To use this mixin in a bundle, declare it like so:
+
+```yaml
+mixins:
+- docker
+```
+
+## Mixin Commands
+The commands available are docker pull, push, build, run, remove, and login. 
+
+## Mixin Syntax
+The same syntax applies for install, upgrade, and uninstall.
+
+### Docker pull
+```yaml
+- docker:
+    description: 
+    pull:
+      name:
+      tag:
+```
+
+### Docker push
+```yaml
+- docker:
+    description: 
+    push:
+      name:
+      tag:
+```
+
+### Docker build
+```yaml
+- docker:
+    description: 
+    build:
+      tag: 
+      file: OPTIONAL
+      path: #defaults to "." OPTIONAL
+```
+
+### Docker run
+```yaml
+- docker:
+    description:
+    run:
+      image:
+      name:
+      detach: #defaults to false
+      ports:
+        - host:
+          container:
+      env:
+        variable: 
+      privileged: #defaults to false
+      rm: #defaults to false
+```
+
+### Docker remove
+```yaml
+- docker:
+    description: 
+    remove:
+      container: 
+      force: #defaults to false
+```
+
+### Docker login
+```yaml
+- docker:
+    description:
+    login: 
+      username: OPTIONAL
```

Comment:
Let's add an explanation of why it is optional, i.e. that it will default to the environment variable provided by DOCKER_USERNAME/PASSWORD from a parameter or credential.